### PR TITLE
improve French sentence with verb modifier

### PR DESCRIPTION
**Pull request by blegaut**
Thanks for contributing!

## Please check if the PR fulfills these requirements 👌:

- [x] Confirm the PR related to a dedicated issue
- [x] Confirm your PR corrects a single bug or implements a single feature
- [x] Your code is linted locally prior to submission
- [x] Your code is fully tested: 99% to 100% coverage
- [ ] Everything is correct on [Sonar dashboard](https://sonarcloud.io/dashboard?id=RosaeNLG_RosaeNLG)
- [x] Confirm your code is under Apache 2.0 license
- [x] Confirm your documentation is under CC-BY-4.0 license
- [x] Confirm license and copyright content is created/updated in each file (see `CONTRIBUTING.md`)
- [ ] Confirm `changelog.adoc` is updated
- [ ] If corrects vulnerabilities, confirm `changelog.adoc` contains CVE IDs

Also remember: each commit **MUST** contain a sign off message (see `CONTRIBUTING.md`) 🙄

## Indicate related issue: 

issue #185 

## Explain what is done, scope, limitations etc.

Mostly add the ability to insert a modifier within the French verb package and the sentence mixin

Still miss documentation and changelog. Tell me what you suggest

## Does this PR introduce a breaking change? 😁
No

### CODE REVIEW

File: packages/rosaenlg-lib/src/LanguageFrench.ts [hunk 1]
```diff
@@ -341,9 +345,11 @@ export class LanguageFrench extends LanguageImpl {
       }
 
       if (sentenceParams.negative) {
-        modifiedVerbalGroup.negativeAdverb = sentenceParams.negativeAdverb || 'pas';
+        modifiedVerbalGroup.negativeAdverb = sentenceParams.negativeAdverb ?? 'pas';
```
Comment (blegaut) [inline packages/rosaenlg-lib/src/LanguageFrench.ts:348]: This allow to pass an empty string as negativeAdverb

File: packages/rosaenlg-lib/test/pug/fr_FR/sentence_negative.pug [hunk 2]
```diff
@@ -202,6 +208,65 @@ mixin SUJET_MP_ref(obj, params)
...
+    verbalGroup: {
+      verb: 'fournir',
+      tense: 'PRESENT',
+      aux:'AVOIR',
+    },
+    objGroups: [
+      {
+        type: 'DIRECT',
+        obj: COD_AIDE,
+        params : { REPRESENTANT: 'ref' }
```
Comment (ludans) [inline packages/rosaenlg-lib/test/pug/fr_FR/sentence_negative.pug:260]: this is a good trick
can you please document it in `sentences.adoc`?

File: packages/rosaenlg-lib/test/pug/fr_FR/sentence_negative.pug [hunk 3]
```diff
@@ -202,6 +208,65 @@ mixin SUJET_MP_ref(obj, params)
...
+  | le jeune aventurier
+  - setRefGender(COI_AVENTURIER, 'aventurier');
+  - setRefNumber(COI_AVENTURIER, 'S');
+- COI_AVENTURIER.ref = COI_AVENTURIER_ref;
+
+-
+  const sentenceTyran = {
+    subjectGroup: {
+      subject: SUJET_TYRAN,
+      params : { deprel: 'subj' }
```
Comment (ludans) [inline packages/rosaenlg-lib/test/pug/fr_FR/sentence_negative.pug:246]: Here we test 2 things:
- modifierAdverb
- deprel

I suggest to make two distinct test cases.

File: packages/rosaenlg-lib/test/pug/fr_FR/sentence_negative.pug [hunk 4]
```diff
@@ -202,6 +208,65 @@ mixin SUJET_MP_ref(obj, params)
     },
   };
 
+//-  Le tyran n'a vraiment fourni aucune aide au jeune aventurier.
+
+
+- const SUJET_TYRAN = {};
+mixin SUJET_TYRAN_ref(obj, params)
```
Comment (ludans) [inline packages/rosaenlg-lib/test/pug/fr_FR/sentence_negative.pug:215]: It is a good example. Can you add it in `sentences_french.adoc`?

File: packages/rosaenlg-lib/src/ValueManager.ts [hunk 5]
```diff
@@ -52,6 +52,7 @@ export interface ValueParams {
   useTheWhenPlural?: boolean; // when a definite determiner and plural, en_US only
   FORCE_DES?: boolean; // French only
   possForm?: PossForm;
+  deprel?: string; //for compatibility use with https://universaldependencies.org/u/dep/index.html when we want to change the anaphor depending of the dependency relation
```
Comment (ludans) [inline packages/rosaenlg-lib/src/ValueManager.ts:55]: Can you please document it?

File: packages/french-verbs/test/conjugation.js
Comment (ludans) [inline packages/french-verbs/test/conjugation.js:1]: please also update french-verbs/README.md

File: packages/rosaenlg-lib/src/LanguageFrench.ts [hunk 6]
```diff
@@ -301,6 +305,7 @@ export class LanguageFrench extends LanguageImpl {
 
     // for pronouns, order is static: COD then COI
     const triggeredList = objGroups
+      .filter((objGroup) => objGroup.params?.REPRESENTANT !== 'ref')
```
Comment (ludans) [inline packages/rosaenlg-lib/src/LanguageFrench.ts:308]: You may also allow to do the contrary, using `REPRESENTANT: 'refexpr'`

File: packages/rosaenlg-lib/src/LanguageFrench.ts [hunk 7]
```diff
@@ -341,9 +345,11 @@ export class LanguageFrench extends LanguageImpl {
       }
 
       if (sentenceParams.negative) {
-        modifiedVerbalGroup.negativeAdverb = sentenceParams.negativeAdverb || 'pas';
+        modifiedVerbalGroup.negativeAdverb = sentenceParams.negativeAdverb ?? 'pas';
       }
 
+      modifiedVerbalGroup.modifierAdverb = sentenceParams.modifierAdverb;
```
Comment (ludans) [inline packages/rosaenlg-lib/src/LanguageFrench.ts:351]: raise an error when `modifierAdverb` is used in English (in `LanguageEnglish.ts`)

File: packages/rosaenlg-lib/src/LanguageFrench.ts (hunk 1 above)
Comment (ludans) [reply packages/rosaenlg-lib/src/LanguageFrench.ts:348]: Yes. Please add an explicit comment in the code so that we don't change it in the future.

**Review by ludans — CHANGES_REQUESTED** [review]
Nice PR, thanks.

Can you please document `modifierAdverb` in `sentences_french.adoc`?

File: packages/rosaenlg-lib/src/LanguageFrench.ts (hunk 1 above)
Comment (blegaut) [reply packages/rosaenlg-lib/src/LanguageFrench.ts:348]: done

File: packages/rosaenlg-lib/test/pug/fr_FR/sentence_negative.pug (hunk 3 above)
Comment (blegaut) [reply packages/rosaenlg-lib/test/pug/fr_FR/sentence_negative.pug:246]: done

File: packages/rosaenlg-lib/test/pug/fr_FR/sentence_negative.pug (hunk 4 above)
Comment (blegaut) [reply packages/rosaenlg-lib/test/pug/fr_FR/sentence_negative.pug:215]: added a similar one.

File: packages/french-verbs/test/conjugation.js
Comment (blegaut) [reply packages/french-verbs/test/conjugation.js:1]: done

File: packages/rosaenlg-lib/src/LanguageFrench.ts (hunk 6 above)
Comment (blegaut) [reply packages/rosaenlg-lib/src/LanguageFrench.ts:308]: I believe that it is easier to declare that you don't want pronoun.

File: packages/rosaenlg-lib/test/pug/fr_FR/sentence_negative.pug (hunk 2 above)
Comment (blegaut) [reply packages/rosaenlg-lib/test/pug/fr_FR/sentence_negative.pug:260]: done

File: packages/rosaenlg-doc/doc/modules/mixins_ref/pages/sentences.adoc [hunk 8]
```diff
@@ -19,3 +19,85 @@ The current support of RosaeNLG for complete sentences is limited:
 * Coverage is *limited* - which means in practice that some structures and combinations will not work
 * The behavior of the `sentence` mixin is not completely aligned with the existing mixins like `verb` or `subjectVerb`
 
+== Useful tricks
+
+In the `sentence` mixin you can pass the usual parameters for the different group, in particular :
+* you can force the referential expression using the parameter `REPRESENTANT: 'ref'`.
+* you can convey the grammatical function using the parameter `deprel`.
```
Comment (ludans) [inline packages/rosaenlg-doc/doc/modules/mixins_ref/pages/sentences.adoc:26]: maybe put your link indicating what 'deprel' means, where it comes from?

File: packages/rosaenlg-doc/doc/modules/mixins_ref/pages/value.adoc [hunk 9]
```diff
@@ -38,6 +38,8 @@ You have to define how to transform the structured object into text.
 
 See xref:referring_expression.adoc[referring expressions].
 
+When integrating rosae with typescript, you can also use the additional parameter `deprel` to pass information about the grammatical function of the object to the referring expressions mixins: for instance, you can restrict the use of the pronoun `he` or `she` to the subject of a sentence.
```
Comment (ludans) [inline packages/rosaenlg-doc/doc/modules/mixins_ref/pages/value.adoc:41]: Maybe indicate that the value of this parameter has to be tested in the mixin that generates the text (i.e., there is not automatic underlying behavior).

**Review by ludans — CHANGES_REQUESTED** [review]
Thanks for the changes and the documentation.

I made a few very last comments.

One last thing: can you please raise an error when `modifierAdverb` is used in English (in `LanguageEnglish.ts`)?

**Review by ludans — APPROVED** [review]
perfect, thanks!

**Pull request merged**